src/views/List/List.jsx

```diff
@@ -51,46 +51,37 @@ export default function List({ token }) {
 
   const onChange = async (listItem) => {
     //when an item is checked, we map through the data array and update the React state
-    const {
-      isActive,
-      id,
-      frequency,
-      lastPurchasedAt = Date.now(),
-      timesPurchased,
-      daysSinceLastPurchase,
-    } = listItem;
-    console.log('listItem:', listItem);
+    const { isActive, id, frequency, lastPurchasedAt, timesPurchased } =
+      listItem;
     const nextData = data.map((item) => {
       if (item.id === id) {
         item.isActive = !item.isActive;
       }
       return item;
     });
     setData(nextData);
-    console.log(nextData);
     //then we use the updateDoc function to update Firebase
     const updatedDoc = doc(db, token, listItem.id);
     const now = Date.now();
     const nextActive = !isActive;
-    // `calculateEstimate` relies on some information that we have to
-    // compute before we can run it.
-    const nextLastPurchasedAt = nextActive ? Date.now() : lastPurchasedAt;
-    const nextDaysSinceLastPurchase = nextActive
-      ? Math.floor((now - lastPurchasedAt) / 86400000) || 1
-      : daysSinceLastPurchase;
-    const nextTimesPurchased = nextActive ? timesPurchased + 1 : timesPurchased;
-
-    const nextFrequency = nextActive
-      ? calculateEstimate(frequency, nextDaysSinceLastPurchase, timesPurchased)
-      : frequency;
-
-    await updateDoc(updatedDoc, {
+    const update = {
       isActive: nextActive,
-      lastPurchasedAt: nextLastPurchasedAt,
-      frequency: nextFrequency,
-      daysSinceLastPurchase: nextDaysSinceLastPurchase,
-      timesPurchased: nextTimesPurchased,
-    });
+    };
+    if (nextActive) {
+      const daysSinceLastPurchase = lastPurchasedAt
+        ? Math.floor((now - lastPurchasedAt) / 86400000)
+        : frequency;
+      update.timesPurchased = timesPurchased + 1;
+      update.lastPurchasedAt = now;
+      update.frequency = calculateEstimate(
+        frequency,
+        daysSinceLastPurchase,
+        update.timesPurchased,
+      );
+      update.timesPurchased = timesPurchased + 1;
+    }
+
+    await updateDoc(updatedDoc, update);
```

Comment:
1️⃣ If the user is checking off the item right now, we need to update more than just the `isActive` state. All of the needed updates are calculated in this if statement.

2️⃣ Line 71 is our solution, and it's a bit hacky.

If there is no `lastPurchasedAt`, we pretend that `frequency` is `daysSinceLastPurchase`. We do this because of (what I think is) a bug in the logic of `calculateEstimate`: If there are < 2 purchases, `calculateEstimate` returns `daysSinceLastPurchase` as the next frequency. If `daysSinceLastPurchase` is a small number, like `0` or `1`, this can really throw off calculations. By setting `daysSinceLastPurchase` to the frequency selected by the user, we can calculate the estimate at a scale that makes a little more sense.

3️⃣ Finally, we send this update object to the database.

Reply:
This makes sense! The only thing I have questions about  is lines 74/82: they are the same but it's working the way it is supposed to. How is that? (i.e., it's not adding to `timesPurchased` 2x)

Reply:
Oh! This is a misrake on my part – those lines are redundant! It works the way it's supposed to becayse it references the same original value each time, and overwrites the key on the update object. e.g., if `timesPurchased` is `1`, we’re setting the key on the object to `2` twice.